omniqueue/src/backends/sqs.rs

```diff
@@ -257,19 +274,42 @@ impl QueueConsumer for SqsQueueConsumer {
         out.messages()
             .unwrap_or_default()
             .iter()
-            .map(|message| -> Result<Delivery, QueueError> {
-                Ok(Delivery {
-                    decoders: self.bytes_registry.clone(),
-                    acker: Box::new(SqsAcker {
-                        ack_client: self.client.clone(),
-                        queue_dsn: self.queue_dsn.clone(),
-                        receipt_handle: message.receipt_handle().map(ToOwned::to_owned),
-                        has_been_acked_or_nacked: false,
-                    }),
-                    payload: Some(message.body().unwrap_or_default().as_bytes().to_owned()),
-                })
-            })
+            .map(|message| -> Result<Delivery, QueueError> { Ok(self.wrap_message(message)) })
             .next()
             .ok_or(QueueError::NoData)?
     }
+
+    async fn receive_all(
+        &mut self,
+        max_messages: usize,
+        deadline: Duration,
+    ) -> Result<Vec<Delivery>, QueueError> {
+        // Ensure that there's at least one message before returning regardless of timeout
```

Comment:
"// Ensure that there's at least one message before returning regardless of timeout"

Is this the behavior we want for receive_all?

If my goal is maximize throughput without overloading my consumer, and I was using this library, I would want `receive_all(max_messages, deadline)` to be
1. Block until at least one message is available. If multiple messages are available, return the batch.
2. If `max_messages.is_some()`, don't return more than `max_messages`
3. If `deadline.is_some()`, don't block longer than the deadline, even if no message are available

(I realize `max_messages` and `deadline` aren't `Option<...>`s in this situation, but maybe they should be?)

The behavior I _wouldn't_ want:
* If only 1 message is immediately available, but `max_messages` > 1, blocking until more messages are ready
* If `deadline` is set, blocking for longer than the deadline if `0` messages are available

Reply:
Check out the updated PR description and diff. Let me know if that meets your expectations.